test/language/arguments-object/mapped/mapped-arguments-nonwritable-nonconfigurable-descriptors-2.js

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2017 Caio Lima. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+info: Mapped arguments property descriptor change to non-writable and non-configurable
+description: >
+    Mapping just stop working when property is set to non-writable. Change the
+    descriptor using [[DefineOwnProperty]] to {writable: false}, set a = 2 and then
+    change property descriptor to {configurable: false}.
+    The descriptor's value is the one set before the property be
+    configured as {writable: false} beacuse mapping was removed.
+flags: [noStrict]
+esid: sec-arguments-exotic-objects-defineownproperty-p-desc
+includes: [propertyHelper.js]
+---*/
+
+function argumentsNonWritableThenNonConfigurableWithInterveningSetMutableBindingDescriptor(a) {
+  Object.defineProperty(arguments, "0", {writable: false});
+  a = 2;
+  Object.defineProperty(arguments, "0", {configurable: false});
+  let propertyDescriptor = Object.getOwnPropertyDescriptor(arguments, '0');
+  assert.sameValue(propertyDescriptor.value, 1);
+  verifyEnumerable(arguments, '0');
+  verifyNotWritable(arguments, '0');
+  verifyNotConfigurable(arguments, '0');
+
+  // Postcondition: Arguments mapping is removed.
+  a = 3;
+  propertyDescriptor = Object.getOwnPropertyDescriptor(arguments, '0');
+  assert.sameValue(propertyDescriptor.value, 1);
+  verifyEnumerable(arguments, '0');
+  verifyNotWritable(arguments, '0');
+  verifyNotConfigurable(arguments, '0');
+}
+argumentsNonWritableThenNonConfigurableWithInterveningSetMutableBindingDescriptor(1);
```

Comment:
this will become a duplicate of `arguments-object/mapped/mapped-arguments-nonwritable-nonconfigurable-descriptors-1.js` if you remove the configurable part.